instrumentation/log4j/log4j-appender-2.17/library/src/main/java/io/opentelemetry/instrumentation/log4j/appender/v2_17/OpenTelemetryAppender.java

```diff
@@ -187,10 +218,32 @@ private static List<String> splitAndFilterBlanksAndNulls(String value) {
    */
   public void setOpenTelemetry(OpenTelemetry openTelemetry) {
     this.openTelemetry = openTelemetry;
+    while (!eventsToReplay.isEmpty()) {
+      try {
+        LogEvent eventToReplay = eventsToReplay.poll(10, TimeUnit.MILLISECONDS);
+        append(eventToReplay);
+      } catch (InterruptedException e) {
+        // Ignore
+      }
+    }
   }
 
+  @SuppressWarnings("SystemOut")
   @Override
   public void append(LogEvent event) {
+    if (openTelemetry == OpenTelemetry.noop()) {
+      if (eventsToReplay.remainingCapacity() > 0) {
+        LogEventToReplay logEventToReplay =
+            new LogEventToReplay(event, captureAllContextDataAttributes);
+        eventsToReplay.offer(logEventToReplay);
+      } else if (!logCacheWarningDisplayed) {
+        logCacheWarningDisplayed = true;
```

Comment:
maybe use AtomicBoolean and

```suggestion
      } else if (!logCacheWarningDisplayed.getAndSet(true)) {
```